src/elements/common/droppable/__tests__/makeDroppable.test.tsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import makeDroppable from '../makeDroppable';
+
+// Test component that will be wrapped with makeDroppable
+const TestComponent = React.forwardRef<HTMLDivElement, { className?: string; 'data-testid'?: string }>(
+    ({ className, 'data-testid': testId, ...props }, ref) => (
+        <div ref={ref} className={className} data-testid={testId || 'test-component'} {...props}>
+            Test Content
+        </div>
+    ),
+);
+
+TestComponent.displayName = 'TestComponent';
+
+describe('elements/common/droppable/makeDroppable', () => {
+    const mockDropValidator = jest.fn();
+    const mockOnDrop = jest.fn();
+    const createDT = () => (typeof DataTransfer === 'function' ? new DataTransfer() : ({} as DataTransfer));
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('Basic functionality', () => {
+        test('should render wrapped component with correct props', () => {
+            const DroppableComponent = makeDroppable({
+                dropValidator: mockDropValidator,
+                onDrop: mockOnDrop,
+            })(TestComponent);
+
+            render(<DroppableComponent className="test-class" data-testid="droppable" />);
```

Comment:
try a `renderComponent` pattern (see other usages in the repo) so that you don't have to redefine these for each test:

```
const renderComponent = (props = {}) => {
    const DroppableComponent = makeDroppable({
        dropValidator: mockDropValidator,
        onDrop: mockOnDrop,
    })(TestComponent);
    return render(<DroppableComponent {...props} />);
};
```

Reply:
you can define `TestComponent` directly in this function too